tests/src/test/scala/cats/tests/XorTTests.scala

```diff
@@ -23,4 +27,34 @@ class XorTTests extends CatsSuite {
       xort.withValidated(_.bimap(f, g)) == xort.bimap(f, g)
     }
   })
+
+  test("fromTryCatch catches matching exceptions") {
+    assert(XorT.fromTryCatch[Option, NumberFormatException]("foo".toInt).isInstanceOf[XorT[Option, NumberFormatException, Int]])
+  }
+
+  test("fromTryCatch lets non-matching exceptions escape") {
+    val _ = intercept[NumberFormatException] {
+      XorT.fromTryCatch[Option, IndexOutOfBoundsException]{ "foo".toInt }
+    }
+  }
+
+  test("fromXor")(check {
+    forAll { (xor: Xor[String, Int]) =>
+      Some(xor.isLeft) == XorT.fromXor[Option](xor).isLeft
+    }
+  })
+
+  implicit val arbitraryTryInt: Arbitrary[Try[Int]] = Arbitrary {
```

Comment:
We're starting to repeat `Arbitrary[Try[_]]`.  While we wait on rickynils/scalacheck#170, should we centralize that somewhere for our own use?